### PR TITLE
Return empty array instead of null when permissions are omitted

### LINKED ISSUE #7247: Roles creation/update responses incorrectly set `null` permissions when not provided.
## This Bug Report affects these Traffic Control components:
- Traffic Ops

## Current behavior:
When the client creates or updates a Role, they may omit the `permissions` property, or explicitly set it to `null`. This is implicitly the same as an empty collection, as further GET requests will reveal. However, in those cases, the immediate response to the PUT or POST request in question will show `permissions` as `null`.

## Expected behavior:
Representations should be consistent, an empty collection is properly represented as an empty collection, not `null`.

## Steps to reproduce:
1. Create or Update a Role, omitting the `permissions` property thereof in the request payload.

### PR DESCRIPTION
**Pull request by AbdelrahmanElawady**
Fixes #7247.

Return empty array instead of `null` in when no permissions are given using `POST` and `PUT` requests to `/roles` endpoint.

<hr/>

## Which Traffic Control components are affected by this PR?
- Traffic Ops

## What is the best way to verify this PR?
using development environment:
1. Login to Traffic Ops
1. Make a `POST` request to `/roles` endpoint with empty permissions
1. Make a `PUT` request to `/roles` with existing role with empty permissions

## If this is a bugfix, which Traffic Control versions contained the bug?
unknown

## PR submission checklist
- [ ] This PR has tests
- [x] This PR has documentation
- [x] This PR has a CHANGELOG.md entry
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Comment by ocket8888** [comment]
This PR should have a CHANGELOG entry and a test

**Comment by AbdelrahmanElawady** [comment]
@ocket8888 I think with the current implementation of TO session for integration tests we can't get response body from roles endpoint for any request methods other than GET. 
https://github.com/apache/trafficcontrol/blob/master/traffic_ops/testing/api/v5/roles_test.go#L220-L247
I am not sure how exactly I could add tests in this case, any suggestion?

**Comment by zrhoffman** [comment]
> @ocket8888 I think with the current implementation of TO session for integration tests we can't get response body from roles endpoint for any request methods other than GET. https://github.com/apache/trafficcontrol/blob/master/traffic_ops/testing/api/v5/roles_test.go#L220-L247 I am not sure how exactly I could add tests in this case, any suggestion?

You can still add tests for testing the roles body, that will just involve adding and checking separate GET request after the POST.

**Comment by AbdelrahmanElawady** [comment]
> You can still add tests for testing the roles body, that will just involve adding and checking separate GET request after the POST.

As mentioned in the issue the problem is with the immediate response from PUT and POST requests. Later GET requests have no problems and not changed in this PR. So how exactly separate GET requests could be used?

**Comment by zrhoffman** [comment]
> > You can still add tests for testing the roles body, that will just involve adding and checking separate GET request after the POST.
> 
> As mentioned in the issue the problem is with the immediate response from PUT and POST requests. Later GET requests have no problems and not changed in this PR. So how exactly separate GET requests could be used?

Indeed, that approach won't help with test coverage here, and restructuring the TO client API library is an improvement outside the scope of this PR.

Rather than adding TO API tests for #7521, adding [unit tests](https://github.com/apache/trafficcontrol/blob/d022dd84c951/traffic_ops/traffic_ops_golang/role/roles_test.go) would be more appropriate.

File: traffic_ops/traffic_ops_golang/role/roles_test.go [hunk 1]
```diff
@@ -20,15 +20,26 @@ package role
...
 	"github.com/apache/trafficcontrol/lib/go-tc"
 	"github.com/apache/trafficcontrol/lib/go-util"
 	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/auth"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/config"
 	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/test"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/trafficvault"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/trafficvault/backends/disabled"
+	"github.com/jmoiron/sqlx"
+	"gopkg.in/DATA-DOG/go-sqlmock.v1"
```
Comment (zrhoffman) [inline traffic_ops/traffic_ops_golang/role/roles_test.go:42]: 3rd-party imports should be in a different import group

**Review by zrhoffman — CHANGES_REQUESTED** [review]
Thanks! Just 1 nit about the import format.

**Review by zrhoffman — APPROVED** [review]
Thanks! Looks great, and tests pass.

**Pull request merged**